aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -668,6 +743,28 @@ module AptosFramework::Stake {
             abort Errors::invalid_argument(EWITHDRAW_NOT_ALLOWED)
         };
 
+        // Update the amount of stake being withdrawn from the active validator set's total voting power.
+        // We only do this if the validator is active.
+        // If the validator is pending_active, it's not possible to unlock immediately if we assume that the min lockup
+        // duration is longer than the epoch duration. So once they have locked up and called join_validator_set, it's
+        // impossible to unlock stake before the epoch ends and they get added to the validator set. Therefore, we don't
+        // need to update anything.
+        // Otherwise if the validator is pending_inactive or inactive, leave_validator_set already deducted all of their
+        // active stake, so we don't need to update anything.
+        let validator_set = borrow_global<ValidatorSet>(@CoreResources);
+        let validator_set_config = borrow_global<ValidatorSetConfiguration>(@CoreResources);
+        let validator_set_voting_power = borrow_global_mut<ValidatorSetVotingPower>(@CoreResources);
+        if (Option::is_some(&find_validator(&validator_set.active_validators, pool_address))) {
+            // If the validator is still active, we do not allow unlocking if remaining stake is below
+            // the min stake required. The validator would need to leave the validator set first.
```

Comment:
we have to make sure the minimum stake is always <= withdraw limit otherwise all of them are deadlocked